tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <concepts>
+#include <iterator>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+struct instantiator {
+    template <input_or_output_iterator Iter>
+    static constexpr void call() {
+        if constexpr (is_permissive) { // TRANSITION, FIXME (File an issue to investigate this later?)
+            (void) input_iterator<Iter>;
+        }
+
+        int input[5] = {1, 2, 3, 4, 5};
+        // [counted.iter.const]
+        {
+            [[maybe_unused]] counted_iterator<Iter> defaultConstructed{};
+
+            [[maybe_unused]] counted_iterator<Iter> constructed(Iter{input}, iter_difference_t<Iter>{2});
+            [[maybe_unused]] counted_iterator<Iter> constructedEmpty{Iter{input}, iter_difference_t<Iter>{0}};
+
+            //[[maybe_unused]] counted_iterator<Iter> convertibleConstructed{};
+            //[[maybe_unused]] counted_iterator<Iter> convertibleAssigned{};

```

Comment:
Should we make `test::iterator<Cat, Elem, Things...>` convertible to `test::iterator<Cat, const Elem, Things...>` so we can test iterator adapter conversions?

Reply:
That was the reason I left it commented so that we can have a discussion on how to best enable the conversion testing. 

Your idea with the conversion to `const Elem` seems to be the best possibility.

Reply:
Done